code/modules/codex/entries/weapons.dm

```diff
@@ -1,3 +1,211 @@
+/obj/item/attachable/get_antag_info()
+	var/list/entries = SScodex.retrieve_entries_for_string(name)
+	var/datum/codex_entry/general_entry = LAZYACCESS(entries, 1)
+	if(general_entry && general_entry.antag_text)
+		return general_entry.antag_text
+
+/obj/item/attachable/get_lore_info()
+	var/list/entries = SScodex.retrieve_entries_for_string(name)
+	var/datum/codex_entry/general_entry = LAZYACCESS(entries, 1)
+	if(general_entry && general_entry.lore_text)
+		return general_entry.lore_text
+
+/obj/item/attachable/get_mechanics_info()
+	. = ..()
+	var/list/attach_strings = list()
+
+	var/list/entries = SScodex.retrieve_entries_for_string(name)
+	var/datum/codex_entry/general_entry = LAZYACCESS(entries, 1)
+	if(general_entry && general_entry.mechanics_text)
+		attach_strings += general_entry.mechanics_text
+
+	if(slot == "muzzle")
+		attach_strings += "This attaches to the muzzle slot on most weapons.<br>"
+	if(slot == "rail")
+		attach_strings += "This attaches to the rail slot on most weapons.<br>"
+	if(slot == "stock")
+		attach_strings += "This attaches to the stock slot on most weapons.<br>"
+	if(slot == "under")
+		attach_strings += "This attaches to the underbarrel slot on most weapons.<br>"
+
+	if(flags_attach_features & ATTACH_REMOVABLE)
+		attach_strings += "This can be field stripped off the weapon if needed."
+
+	if(flags_attach_features & ATTACH_ACTIVATION)
+		attach_strings += "This needs to be activated to be used."
+
+	if(flags_attach_features & ATTACH_RELOADABLE)
+		attach_strings += "This can be reloaded with the appropriate ammunition."	
+
+	//if(max_rounds) doesn't work for all the attachements, only the attached_gun ones, UGL, shottie, flamer.
```

Comment:
You should then attempt to make it work with those, what seems to be the issue exactly?